ext/node/lib.rs

```diff
@@ -44,13 +44,13 @@ pub trait NodeEnv {
 }
 
 pub trait NodePermissions {
-  fn check_read(&mut self, path: &Path) -> Result<(), AnyError>;
+  fn check_read(&self, path: &Path) -> Result<(), AnyError>;
```

Comment:
We may want to do this across the board with all permissions traits?

Reply:
I believe we did this to signal that permissions can be mutated, but since it's interior mutability I don't think that the callers should be concerned about it. So +1 from me

Reply:
I think it’s because previously we didn’t have the PermissionContainer. It’s probably best to force everyone to use interior mutability in order to prevent bugs like what we saw before we had that.